src/ConservationChecker.jl

```diff
@@ -0,0 +1,331 @@
+
+    # save radiation source
+    if radiation != nothing
+        face_space = axes(atmos_sim.integrator.u.f)
+        z = parent(Fields.coordinate_field(face_space).z)
+        Δz_top = round(FT(0.5) * (z[end, 1, 1, 1, 1] - z[end - 1, 1, 1, 1, 1]))
+        n_faces = length(z[:, 1, 1, 1, 1])
+
+        LWd_TOA = Fields.level(
+            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_lw_flux_dn), face_space),
+            n_faces - half,
+        )
+        LWu_TOA = Fields.level(
+            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_lw_flux_up), face_space),
+            n_faces - half,
+        )
+        SWd_TOA = Fields.level(
+            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_sw_flux_dn), face_space),
+            n_faces - half,
+        )
+        SWu_TOA = Fields.level(
+            RRTMGPI.array2field(FT.(atmos_sim.integrator.p.radiation_model.face_sw_flux_up), face_space),
+            n_faces - half,
+        )
+
+        coupler_sim.fields.F_R_TOA .-=
+            swap_space!(LWd_TOA .+ SWd_TOA .- LWu_TOA .- SWu_TOA, boundary_space) .* coupler_sim.Δt_cpl
+        radiation_sources_accum = sum(coupler_sim.fields.F_R_TOA) # accumulated radiation sources + sinks [J]
+        push!(cc.toa_net_source, radiation_sources_accum)
+    else
+        push!(cc.toa_net_source, FT(0))
+    end
+
+    # save atmos
+    push!(cc.ρe_tot_atmos, sum(atmos_sim.integrator.u.c.ρe_tot))
+
+    # save land
+    if land_sim !== nothing
+        e_per_area_land = zeros(axes(land_sim.integrator.u.bucket.W))
+        get_land_energy(land_sim, e_per_area_land)
+        push!(cc.ρe_tot_land, sum(e_per_area_land .* surface_masks.land))
+    else
+        push!(cc.ρe_tot_land, FT(0))
+    end
+
+    # save sea ice
+    if ice_sim !== nothing
+        push!(cc.ρe_tot_seaice, sum(get_slab_energy(ice_sim, ice_sim.integrator.u.T_sfc) .* surface_masks.ice))
+    else
+        push!(cc.ρe_tot_seaice, FT(0))
+    end
+
+    # save ocean
+    if ocean_sim !== nothing
+        push!(cc.ρe_tot_ocean, sum(get_slab_energy(ocean_sim, ocean_sim.integrator.u.T_sfc) .* surface_masks.ocean))
+    else
+        push!(cc.ρe_tot_ocean, FT(0))
+    end
+
+    # save surface friction sink
+    include_ke_dissipation = false # NB: currently, kinetic energy dissipation is neglected in total energy tendency
+    if include_ke_dissipation
+        push!(cc.friction_sink, sum((ke_dissipation(atmos_sim)) .* coupler_sim.Δt_cpl)) # ρ d ke_friction / dt 
+    else
+        push!(cc.friction_sink, FT(0))
+    end
+
+    push!(cc.ice_base_source, FT(0))
+
+    tot =
+        @. cc.ρe_tot_atmos + cc.ρe_tot_ocean + cc.ρe_tot_land + cc.ρe_tot_seaice + cc.toa_net_source + cc.friction_sink
+    # @assert abs((tot[end] - tot[1]) / tot[end]) < 1e-4
+end
+
+"""
+    ke_dissipation(sim)
+
+Calculate the kinetic energy diffipation of a simulation. 
+"""
+function ke_dissipation(sim)
```

Comment:
could be helpful to annotate the input and return types (at least in the docstring)

Reply:
This function is now removed.